clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java

```diff
@@ -683,8 +685,8 @@ public void onSuccess(ClientResponse response, RequestFuture<Map<TopicPartition,
      * @param listOffsetResponse The response from the server.
      * @param future The future to be completed when the response returns. Note that any partition-level errors will
      *               generally fail the entire future result. The one exception is UNSUPPORTED_FOR_MESSAGE_FORMAT,
-     *               which indicates that the broker does not support the v1 message format and results in a null
-     *               being inserted into the resulting map.
+     *               which indicates that the broker does not support the v1 message format and causes a null value
```

Comment:
I'm trying to understand how this change would work. One difference between `OffsetData` and `OffsetAndTimestamp` is that `OffsetData` supports null timestamps (its `timestamp` type is `Long`, whereas in `OffsetAndTimestamp` the type is `long`). And in the method we may need to set the timestamp to `null` (i.e. for v0 response).

Also, you suggested replacing the loop with a `putAll`. How would partition-level error handling (and raising the appropriate exception) work if we get rid of the loop?

Thanks for clarifying.